lib/vagrant-service-manager/binary_handlers/binary_handler.rb

```diff
@@ -146,6 +147,19 @@ def validate_url
 
       private
 
+      def set_proxy_environment
+        return if @machine.config.servicemanager.proxy.nil?
+
+        env_proxy = @machine.config.servicemanager.proxy
+        if @machine.config.servicemanager.proxy_user
+          user = @machine.config.servicemanager.proxy_user
+          password = @machine.config.servicemanager.proxy_password
```

Comment:
So what if password is empty? Does it work? Is this still the required syntax? Personally, I'd decouple the creation of the proxy string from setting it. You make it harder to test. If they are separate function you can test just the creation of the _env_proxy_ with various values. A true unit test. This test does both things together and make testing hard again.